src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -640,4 +643,72 @@ public void testOpenHtmlContentWithCurrentWebDriver()
         SelenideAddons.openHtmlContentWithCurrentWebDriver(textHtml);
         Assert.assertEquals(text, $("body").getText());
     }
+
+    private int customWaitingTime = 3000;
+
+    @Test
+    public void testOptionalWaitUntil()
+    {
+        openBlogPage();
+        SelenideElement privaceDialog = $("#privacy-message");
+        boolean isVisible = SelenideAddons.optionalWaitUntilCondition(privaceDialog, visible, customWaitingTime);
+        assertTrue("the privacy message dialog was not found within the timeframe", isVisible);
+        long startTime = new Date().getTime();
+        boolean isHidden = SelenideAddons.optionalWaitUntilCondition(privaceDialog, hidden, customWaitingTime);
+        long endTime = new Date().getTime();
+        assertFalse("the privacy message dialog was unexpectedly hidden during the timeframe", isHidden);
+        int waitingTime = customWaitingTime * Neodymium.configuration().optionalElementRetryCount();
+        Assert.assertTrue("Runtime was shorter than expected", Range.between(waitingTime, waitingTime + customWaitingTime)
+                               .contains(Math.toIntExact(endTime - startTime)));
+    }
+
+    @Test
+    public void testOptionalWaitWhile()
+    {
+        openBlogPage();
+        SelenideElement privaceDialog = $("#privacy-message").shouldBe(visible);
+        long startTime = new Date().getTime();
+        boolean isHidden = SelenideAddons.optionalWaitWhileCondition(privaceDialog, visible, customWaitingTime);
+        long endTime = new Date().getTime();
+        assertFalse("the privacy message dialog was unexpectedly during within the timeframe", isHidden);
```

Comment:
```suggestion
        assertFalse("the privacy message dialog was unexpectedly hidden within the timeframe", isHidden);
```

Reply:
Don't you agree with the suggested changes? Why?

Reply:
Sorry, did not see. Should be in now.